client/map_verifier.go

```diff
@@ -81,7 +82,10 @@ func (m *MapVerifier) VerifyMapLeafInclusion(smr *trillian.SignedMapRoot, leafPr
 
 // VerifyMapLeafInclusionHash verifies a MapLeafInclusion object against a root hash.
 func (m *MapVerifier) VerifyMapLeafInclusionHash(rootHash []byte, leafProof *trillian.MapLeafInclusion) error {
-	return merkle.VerifyMapInclusionProof(m.MapID, leafProof.GetLeaf(), rootHash, leafProof.GetInclusion(), m.Hasher)
+	if err := merkle.VerifyMapInclusionProof(m.MapID, leafProof.GetLeaf(), rootHash, leafProof.GetInclusion(), m.Hasher); err != nil {
+		return status.Errorf(status.Code(err), "VerifyMapLeafInclusionHash(): %v, trillian.MapLeafInclusion=\n%v", err, proto.MarshalTextString(leafProof))
```

Comment:
Why not make the caller of this function do the error wrapping?

Reply:
Sometimes I prefer an error format more like
`VerifyMapLeafInclusionHash(<args>): err`

Agreed that putting this in the client library would make it more useful.